src/pymor/reductors/ph/basic.py

```diff
@@ -0,0 +1,67 @@
+from pymor.algorithms.projection import project, project_to_subbasis
+from pymor.models.iosys import PHLTIModel
+from pymor.reductors.basic import ProjectionBasedReductor
+
+
+class PHLTIPGReductor(ProjectionBasedReductor):
+    """Petrov-Galerkin projection of an |PHLTIModel|.
+
+    Parameters
+    ----------
+    fom
+        The full order |PHLTIModel| to reduce.
+    V
+        The basis of the ansatz space.
+    E_biorthonormal
+        If `True`, no `E` matrix will be assembled for the reduced |Model|.
+        Set to `True` if `W` and `V` are biorthonormal w.r.t. `fom.E`.
+    """
+
+    def __init__(self, fom, V, E_biorthonormal=False):
+        assert isinstance(fom, PHLTIModel)
+
+        W = self._obtain_W(fom, V)
+
+        super().__init__(fom, {'W': W, 'V': V})
+        self.E_biorthonormal = E_biorthonormal
+
+    def project_operators(self):
+        fom = self.fom
+        W = self.bases['W']
+        V = self.bases['V']
+        projected_operators = {'E': None if self.E_biorthonormal else project(fom.E, W, V),
+                               'J': project(fom.J, W, V),
+                               'R': project(fom.R, W, V),
+                               'G': project(fom.G, W, None),
+                               'P': project(fom.P, W, None),
+                               'S': fom.S,
+                               'N': fom.N
+                               }
+        return projected_operators
+
+    def project_operators_to_subbasis(self, dims):
+        if dims['W'] != dims['V']:
+            raise ValueError
+        rom = self._last_rom
+        dim = dims['V']
+        projected_operators = {'E': None if self.E_biorthonormal else project_to_subbasis(rom.E, dim, dim),
+                               'J': project_to_subbasis(rom.J, dim, dim),
+                               'R': project_to_subbasis(rom.R, dim, dim),
+                               'G': project_to_subbasis(rom.B, dim, None),
+                               'P': project_to_subbasis(rom.P, dim, None),
+                               'S': rom.S,
+                               'N': rom.N}
+        return projected_operators
+
+    def build_rom(self, projected_operators, error_estimator):
+        return PHLTIModel(error_estimator=error_estimator, **projected_operators)
+
+    def extend_basis(self, **kwargs):
+        raise NotImplementedError
+
+    def reconstruct(self, u, basis='V'):
+        return super().reconstruct(u, basis)
+
+    def _obtain_W(self, fom, V):
+        # If with Q, W = QV
```

Comment:
This needs updating?

Reply:
Not in this PR, since in this state we only allow pH systems without $Q$ as PHLTIModel. We add support for pH systems with $Q$ in #1847 and also there I am not sure if we want to update this, since there we just transform a given pH system with $Q$ to a _generalized_ pH system without $Q$ (shift $Q$ on LHS by multiplying with $Q^\top$) by executing the function below.

https://github.com/pymor/pymor/blob/dfe7c32dbb4b92b8128c74837cb65a13e248d970/src/pymor/models/iosys.py#L1479-L1499

I think from a numerical point of view it makes sense to get rid of the matrix product $(J-R)Q$ before executing pHIRKA.

Reply:
I guess transforming the system depends on whether we allow $Q$ to be singular. Or is it necessary for $Q$ to be invertible for the projection with $V$ and $W = Q V$ to make sense?

Then, if we always only do Galerkin projection, maybe this class should be called `PHLTIGalerkinReductor`?

Reply:
Doesn't $Q$ is regular follows from $Q^T E \succ 0$?

I think we can just set $W=QV$ in every case in  #1836 since then we have then $Q$ is `IdentityOperator` or `NumpyMatrixOperator` and I guess applying an `IdentityOperator` to $V$ should not be too costly, or is it? And in the case of $Q\neq I$ we have to do it anyway.

Reply:
I've pushed this proposal in #1836 see

https://github.com/pymor/pymor/blob/443a73e0418269e25117c9148bfb08c3e25d3896/src/pymor/reductors/ph/basic.py#L6-L26

Reply:
> Doesn't $Q$ is regular follows from $Q^T E \suc≻

Aha, ok, I didn't see that change in the refactoring PR.

Then `W = Q V` seems ok to me (if `Q` is the `IdentityOperator`, no computation should be done).